### DISCUSSION TITLE
Prevent Theme Breakage: Check and Clean HTML from WYSIWYG Editor to Avoid Syntax Errors

### DISCUSSION THREAD

**Discussion by empiricompany** [OpenMage/magento-lts#3391] [Ideas]
Sometimes it happens that the customer enters HTML code in the product description or CMS without properly closing some tags, resulting in the theme breaking on the frontend.

One way to avoid these issues would be to clean the text before saving it in the database, for example, using tidy::cleanRepair().
https://www.php.net/manual/en/tidy.cleanrepair.php

We should try different configuration combinations so that it doesn't alter the HTML but only fixes major errors like tag closure.

You could make it an optional requirement on the server, meaning that the clean function is only executed if the function is available.

**Comment by addison74** [comment]
The observation made is fine.

I also mentioned here that in Frontend we have problems with some forms that do not trim what is entered in the text fields and often visitors enter spaces. I came across this in the order form and I had to remove the extra spaces. It is not something serious, but engineering speaking, it can be prevented and gives a good artistic impression.

**Comment by fballiano** [comment]
for sure https://github.com/OpenMage/magento-lts/pull/3220 will help ;-)

**Reply by addison74** [reply]
Only it the user uses the Tiny editor. If he fills in the html content the validation before saving is required.

**Reply by fballiano** [reply]
true
